leveldb tinylicious tests

kurtb opened the pull request:
Adds leveldb tests for tinylicious. Does so by expanding on the previous memory collection tests. By making use of the DbFactory we can target both database types with the same test code. A simple test factory is needed for custom init and cleanup (i.e. deleting the leveldb directory).

kurtb committed:
leveldb tinylicious tests

Adds leveldb tests for tinylicious. Does so by expanding on the previous memory
collection tests. By making use of the DbFactory we can target both database
types with the same test code. A simple test factory is needed for custom init
and cleanup (i.e. deleting the leveldb directory).

tanviraumi commented on server/tinylicious/src/test/dbFactory.spec.ts:
```diff
@@ -0,0 +1,167 @@
+                    return {
+                        value: {
+                            db: {
+                                inMemory: true,
```
false?

kurtb replied:
Ah - good catch! Fixed things up to actually turn leveldb on...

As part of this also needed to tweak the tests a bit. MongoDB is the basis for these DB APIs. And both in memory/level are subsets of the Mongo API. Tinylicious/r11s obviously don't use the full set of functionality.

The test now better captures the usage of each collection and serves as a bit of a conformance test should anyone add another use case.

Having a more constrained DB API may be nice to better highlight the actual required functionality.

tanviraumi reviewed (commented):
Looks good but the in memory config for level db was set to true so the tests are probably only running in memory.

kurtb committed:
dbfactory test updates

fixes a bug where leveldb config was incorrectly not setting inMemory to false.

Also groups tests into functionality groups as used by tinylicious. Both memory/level are subsets of mongo and so this targets the actual behavior used by tinylicious.

tanviraumi reviewed (approved):
Thank you for the refactoring and tests. The tests are more comprehensive than our current usage of levelDb which is great.

tylerbutler commented:
/azp run repo-policy-check

tylerbutler commented:
Our build pipeline is having some issues unrelated to this PR. Assigned this bug to myself to merge once our build issues are resolved.

tylerbutler commented:
Closing and reopening to trigger PR validation.
